common/test-setting-immutable-prototype.js

```diff
@@ -1,24 +1,27 @@
-self.testSettingImmutablePrototypeToNewValueOnly = (prefix, target, newValue, newValueString) => {
+self.testSettingImmutablePrototypeToNewValueOnly =
+  (prefix, target, newValue, newValueString, { isSameOriginDomain }) => {
   test(() => {
     assert_throws(new TypeError, () => {
       Object.setPrototypeOf(target, newValue);
     });
   }, `${prefix}: setting the prototype to ${newValueString} via Object.setPrototypeOf should throw a TypeError`);
 
+  const dunderProtoError = isSameOriginDomain ? { name: "TypeError" } : { name: "SecurityError" };
```

Comment:
This won't check for the "SecurityError" being a DOMException.  How about:

    const dunderProtoError = isSameOriginDomain ? new TypeError() : "SecurityError";

?